staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -831,6 +834,122 @@ func TestTransformationFailure(t *testing.T) {
 	}
 }
 
+func TestKvsSplit(t *testing.T) {
+	testCases := []struct {
+		desc           string
+		kvs            []*mvccpb.KeyValue
+		numberOfChunks int
+		want           [][]*mvccpb.KeyValue
+	}{
+		{
+			desc:           "empty input results in nil - can't split an empty slice",
+			kvs:            []*mvccpb.KeyValue{},
+			numberOfChunks: 4,
+			want:           nil,
+		},
+		{
+			desc:           "input slice on len one",
+			kvs:            []*mvccpb.KeyValue{{Key: []byte("key_one")}},
+			numberOfChunks: 4,
+			want:           [][]*mvccpb.KeyValue{{&mvccpb.KeyValue{Key: []byte("key_one")}}},
+		},
+		{
+			desc: "input slice of len two",
+			kvs: []*mvccpb.KeyValue{
+				{Key: []byte("key_one")},
+				{Key: []byte("key_two")},
+			},
+			numberOfChunks: 4,
+			want: [][]*mvccpb.KeyValue{
+				{
+					&mvccpb.KeyValue{Key: []byte("key_one")},
+				},
+				{
+					&mvccpb.KeyValue{Key: []byte("key_two")},
+				},
+			},
+		},
+		{
+			desc: "input slice on len three",
+			kvs: []*mvccpb.KeyValue{
+				{Key: []byte("key_one")},
+				{Key: []byte("key_two")},
+				{Key: []byte("key_three")},
+			},
+			numberOfChunks: 4,
+			want: [][]*mvccpb.KeyValue{
+				{
+					&mvccpb.KeyValue{Key: []byte("key_one")},
+				},
+				{
+					&mvccpb.KeyValue{Key: []byte("key_two")},
+				},
+				{
+					&mvccpb.KeyValue{Key: []byte("key_three")},
+				},
+			},
+		},
+		{
+			desc: "input slice on len four",
+			kvs: []*mvccpb.KeyValue{
+				{Key: []byte("key_one")},
+				{Key: []byte("key_two")},
+				{Key: []byte("key_three")},
+				{Key: []byte("key_four")},
+			},
+			numberOfChunks: 4,
+			want: [][]*mvccpb.KeyValue{
+				{
+					&mvccpb.KeyValue{Key: []byte("key_one")},
+				},
+				{
+					&mvccpb.KeyValue{Key: []byte("key_two")},
+				},
+				{
+
+					&mvccpb.KeyValue{Key: []byte("key_three")},
+				},
+				{
+					&mvccpb.KeyValue{Key: []byte("key_four")},
+				},
+			},
+		},
+		{
+			desc: "input slice on len five",
+			kvs: []*mvccpb.KeyValue{
+				{Key: []byte("key_one")},
+				{Key: []byte("key_two")},
+				{Key: []byte("key_three")},
+				{Key: []byte("key_four")},
+				{Key: []byte("key_five")},
+			},
+			numberOfChunks: 4,
+			want: [][]*mvccpb.KeyValue{
+				{
+					&mvccpb.KeyValue{Key: []byte("key_one")},
+					&mvccpb.KeyValue{Key: []byte("key_two")},
+				},
+				{
+					&mvccpb.KeyValue{Key: []byte("key_three")},
+					&mvccpb.KeyValue{Key: []byte("key_four")},
+				},
+				{
+					&mvccpb.KeyValue{Key: []byte("key_five")},
+				},
```

Comment:
hmm, why is this not:
- `key_one`, `key_two`
- `key_three`
- `key_four`
- `key_five`

I'd expect 4 chunks, not 3.

Reply:
Done - the new logic produces the expected outcome.